src/taipy/core/data/mongo.py

```diff
@@ -197,9 +194,8 @@ def _insert_dicts(self, data: List[Dict]) -> None:
 
         This method will overwrite the data contained in a list of dictionaries into a collection.
         """
-        write_collection = self.mongo_db[self.properties.get("write_collection")]
-        write_collection.delete_many({})
-        write_collection.insert_many(data)
+        self.collection.delete_many({})
```

Comment:
I'm having mixed feelings as to why we would want to delete everything before we want to write new data to the collection. My point of view as a user would be I want to record my historical results, deleting everything would prevent me from doing any comparison with historical data. And a collection that keeps deleting the document sounds weird. My proposal would be to not delete anything when inserting new things, or add another field to the config for user to specify if they want to overwrite everything when writing or not (by default I strongly think it should be no)

Reply:
For consistency reasons with other data nodes, we should keep the deletion. It is also a way to ensure the consistency of the collection itself. For example, let s imagine we write twice the data with 2 different datasets that have some overlapping. What would be the result?

However, the use case you mentioned is completely valid. Sometimes we want to keep history. But I believe this use case is more generic, and not specific to MongoDB. We should handle this on a different ticket.
For SQL data node we can handle the use case by using SQLDataNode with custom queries and not SQLTableDataNode. We could imagine something similar for Mongo, but I am not sure that is the right way. 
A way to do it for all kinds of data nodes would be to have an overwrite attribute (as you proposed). If it's False, we systematically add a `writer_id` field in the data (mongo document, SQL schema, CSV column, Excel column, ...) that would correspond to the identifier of the writer (basically the JobId). FYI, the JobId is already stored in the data node when we write it.